CMakeLists.txt

```diff
@@ -20,61 +20,43 @@ set(CMAKE_CXX_EXTENSIONS OFF) #...without compiler extensions like gnu++11
 option(CMAKE_VERBOSE_MAKEFILE "Generate verbose Makefiles" OFF)
 include(GNUInstallDirs)
 
+# allow users to directly point to a kokkos install
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
-find_package(Kokkos REQUIRED)
-
-option(Cabana_ENABLE_Serial "Build Cabana with Serial support" ON)
-
-option(Cabana_ENABLE_Pthread "Build Cabana with Pthread support" OFF)
-if( Cabana_ENABLE_Pthread )
-  find_package(Threads)
+set(Kokkos_INSTALL_PATH "" CACHE PATH "Kokkos installation location" )
+if(Kokkos_INSTALL_PATH)
+  find_package(Kokkos 3 PATHS ${Kokkos_INSTALL_PATH} NO_DEFAULT_PATH)
 endif()
-
-option(Cabana_ENABLE_OpenMP "Build Cabana with OpenMP support" OFF)
-if( Cabana_ENABLE_OpenMP )
-  find_package(OpenMP)
+if(NOT Kokkos_FOUND)
+  find_package(Kokkos 3 REQUIRED)
 endif()
 
-option(Cabana_ENABLE_Cuda "Build Cabana with Cuda support" OFF)
-if( Cabana_ENABLE_Cuda )
-  enable_language( CUDA )
-endif()
+# set supported devices
+set(CABANA_SUPPORTED_DEVICES SERIAL THREADS OPENMP CUDA)
 
-set(CABANA_SUPPORTED_DEVICES Serial Pthread OpenMP Cuda)
-set(ENABLE_DEVICES)
+# check user required device types
 foreach(_device ${CABANA_SUPPORTED_DEVICES})
-  if(Cabana_ENABLE_${_device})
-    string(TOUPPER ${_device} _uppercase_device)
-    if(NOT Kokkos_DEVICES MATCHES ".*${_uppercase_device}.*")
-      message(FATAL_ERROR "Kokkos built with devices '${Kokkos_DEVICES}', but Cabana with ${_device} requested.")
-    endif()
-    list(APPEND ENABLE_DEVICES "${_device}")
-  endif()
-endforeach()
-if(NOT ENABLE_DEVICES)
-  message(FATAL_ERROR "NO Cabana backend enabled")
-endif()
-message(STATUS "Enable Devices: ${ENABLE_DEVICES}")
-
-# if we are using Cuda, also add a "CudaUVM" device to use that memory space
-# in the test suite. Also enable a Cabana UVM option so the tests get triggered.
-foreach(_device ${ENABLE_DEVICES})
-  if(_device STREQUAL Cuda)
-    list(APPEND CABANA_SUPPORTED_DEVICES CudaUVM)
-    set(Cabana_ENABLE_CudaUVM ON)
+  option(Cabana_REQUIRE_${_device} "Build Cabana with required Kokkos ${device} support" OFF)
+  if( Cabana_REQUIRE_${_device} )
+    kokkos_check( DEVICES ${_device} )
   endif()
 endforeach()
 
+# enable unit tests
 option(Cabana_ENABLE_TESTING "Build tests" OFF)
 if(Cabana_ENABLE_TESTING)
   enable_testing()
 endif()
 
-find_package(Doxygen)
-if(Doxygen_FOUND)
-  doxygen_add_docs(doxygen core/src)
+# enable doxygen
+option(Cabana_ENABLE_DOXYGEN "Build doxygen" ON)
```

Comment:
Why? One can just use `CMAKE_DISABLE_FIND_PACKAGE_Doxygen=ON` if you really really don't want to detect doxygen.

Reply:
See the discussion above - I have removed this change.